pepper-apis/dsm-core/src/main/java/org/broadinstitute/lddp/security/SecurityHelper.java

```diff
@@ -69,12 +69,12 @@ public static String createToken(@NonNull String secret, long invalidAfter, @Non
         }
     }
 
-    public static String createGpToken(@NonNull String secret, long invalidAfter, @NonNull Map<String, String> claims) {
+    public static String createTokenWithSigner(@NonNull String secret, long invalidAfter, @NonNull Map<String, String> claims, String signer) {
         try {
```

Comment:
Should the signer also be NonNull?

Reply:
No unfortunately, GP does not have one set string that they shared with us to check

Reply:
Not sure I understand. Do we need a check for a null signer in the code?

Reply:
What I meant is, GP tokens come in all shapes and forms, we had to drop some conditions and checks between Mercury and BSP so their tokens get validated

Reply:
It seems like we need to check for an empty/null signer since we are assigning it below.
If @NonNull is not a good choice then we should check for null/empty before assigning the signer.
If a empty/null signer is valid, then all we need is a comment saying so, since that is unusual.